cmd/extproc/mainlib/main.go

```diff
@@ -85,6 +91,42 @@ func parseAndValidateFlags(args []string) (extProcFlags, error) {
 	return flags, errors.Join(errs...)
 }
 
+// parseMetricsHeaderNames parses a comma-separated string of header names
+// and returns a slice of sanitized header names for use as metric labels.
+func parseMetricsHeaderNames(headerNamesStr string) []string {
+	if headerNamesStr == "" {
+		return nil
+	}
+
+	headers := strings.Split(headerNamesStr, ",")
+	var result []string
+
+	for _, header := range headers {
+		header = strings.TrimSpace(header)
+		if header != "" {
+			// Convert to lowercase for consistent handling.
+			header = strings.ToLower(header)
+			// Sanitize header name for Prometheus label compatibility.
+			// Replace any non-alphanumeric characters (except underscores) with underscores.
+			sanitized := strings.Map(func(r rune) rune {
+				if (r >= 'a' && r <= 'z') || (r >= '0' && r <= '9') || r == '_' {
+					return r
+				}
+				return '_'
+			}, header)
+
+			// Ensure it doesn't start with a number (Prometheus requirement).
+			if len(sanitized) > 0 && sanitized[0] >= '0' && sanitized[0] <= '9' {
+				sanitized = "header_" + sanitized
+			}
```

Comment:
shouldn't this be happening at much earlier phase: the main routine of the controller at the startup time?